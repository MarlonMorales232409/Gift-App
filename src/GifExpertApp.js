import React, { useState } from "react";
import AddCategory from "./Components/AddCategory";
import GifGrid from "./Components/GifGrid";

const GifExpertApp = () => {
  const [categories, setCategories] = useState([]);

  return (
    <>
      <h2 className="title-app">Gift Expert App</h2>
      <hr />
      <div className="gift-main">
        <AddCategory setCategories={setCategories} />
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </div>
    </>
  );
};

export default GifExpertApp;
