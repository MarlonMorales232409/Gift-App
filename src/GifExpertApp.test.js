import React from "react";
import { shallow } from "enzyme";
import GifExpertApp from "./GifExpertApp";

describe("Should mathc with the component <GifExpertApp/>", () => {
  test("should mathc with the component <GifExpertApp/>", () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should match the snapshopt if exist categories", () => {
    const categories = ["retsuko", "vader"];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
