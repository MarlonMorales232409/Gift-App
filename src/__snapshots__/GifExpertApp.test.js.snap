// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should mathc with the component <GifExpertApp/> should match the snapshopt if exist categories 1`] = `
<Fragment>
  <h2
    className="title-app"
  >
    Gift Expert App
  </h2>
  <hr />
  <div
    className="gift-main"
  >
    <AddCategory
      setCategories={[Function]}
    />
    <GifGrid
      category="retsuko"
      key="retsuko"
    />
    <GifGrid
      category="vader"
      key="vader"
    />
  </div>
</Fragment>
`;

exports[`Should mathc with the component <GifExpertApp/> should mathc with the component <GifExpertApp/> 1`] = `
<Fragment>
  <h2
    className="title-app"
  >
    Gift Expert App
  </h2>
  <hr />
  <div
    className="gift-main"
  >
    <AddCategory
      setCategories={[Function]}
    />
  </div>
</Fragment>
`;
